### PR TITLE
Add CBMC proof of initFileTransfer_Mqtt function

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of initFileTransfer_Mqtt function

Description
-----------
Added CBMC proof files for initFileTransfer_Mqtt function

Checklist:
----------

- [ x] I have tested my changes. No regression in existing tests.
- [ x] My code is formatted using Uncrustify.
- [ x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    return status;
+}
+
+/*********************************************************/
+
+void initFileTransfer_Mqtt_harness()
+{
+    OtaAgentContext_t agent;
+    OtaFileContext_t filecontext;
+    OtaInterfaces_t pOtaInterface;
```
Comment (yanjos-dev) [inline test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c:51]: This variable isn't a pointer, can you remove p from the start of it?

File: test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    /* subscribe reference to the mqtt function is expected to be assigned by the user and thus
+     * assumed not to be NULL. */
+    pOtaInterface.mqtt.subscribe = subscribe;
+    agent.pOtaInterface = &pOtaInterface;
+
+    /* initFileTransfer_Mqtt function is only called when there is a file
+     * ready to be downloaded. Thus filecontext cannot be NULL */
```
Comment (yanjos-dev) [inline test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c:59]:
```suggestion
    /* initFileTransfer_Mqtt function is only called when there is a file
     * ready to be downloaded. Thus filecontext cannot be NULL. */
```

File: test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+/*********************************************************/
+
+void initFileTransfer_Mqtt_harness()
+{
+    OtaAgentContext_t agent;
+    OtaFileContext_t filecontext;
+    OtaInterfaces_t pOtaInterface;
+
+    /* subscribe reference to the mqtt function is expected to be assigned by the user and thus
+     * assumed not to be NULL. */
```
Comment (yanjos-dev) [inline test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c:54]:
```suggestion
    /* Subscribe reference to the MQTT function is expected to be assigned by
     * the user and thus assumed not to be NULL. */
```

File: test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c (hunk 1 above)
Comment (Bujain) [reply test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c]: Updated in f99d77c

File: test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c (hunk 2 above)
Comment (Bujain) [reply test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c]: Updated in f99d77c

File: test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c (hunk 3 above)
Comment (Bujain) [reply test/cbmc/proofs/initFileTransfer_Mqtt/initFileTransfer_Mqtt_harness.c]: Updated in f99d77c

**Comment by yanjos-dev** [comment]
It looks like the memory statistics test is failling. I don't see why it would fail when you haven't modified any of the library files. Can you look into this?

**Review by yanjos-dev — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**